modules/stochastic_tools/src/samplers/Metropolis.C

```diff
@@ -0,0 +1,103 @@
+#include "Normal.h"
+
+registerMooseObjectAliased("StochasticToolsApp", Metropolis, "Metropolis");
+registerMooseObjectReplaced("StochasticToolsApp", Metropolis, "07/01/2020 00:00", Metropolis);
+
+InputParameters
+Metropolis::validParams()
+{
+  InputParameters params = Sampler::validParams();
+  params.addClassDescription("Markov Chain Monte Carlo sampling using Metropolis algorithm.");
+  params.addRequiredParam<dof_id_type>("num_rows", "The number of Markov chains.");
+  params.addRequiredParam<std::vector<DistributionName>>("distributions",
+                                                         "The distribution names to be sampled.");
+  params.addRequiredParam<VectorPostprocessorName>(
+      "inputs_vpp", "Vectorpostprocessor with the samples created by sampler.");
+  params.addRequiredParam<std::vector<std::string>>(
+      "inputs_names",
+      "Name of vector from vectorpostprocessor with the samples created by sampler.");
+  params.addRequiredParam<std::vector<Real>>("proposal_std",
+                                             "Standard deviations of the proposal distributions.");
+  params.addRequiredParam<std::vector<Real>>(
+      "initial_values", "Seed input values to get the Metropolis sampler started.");
+  return params;
+}
+
+Metropolis::Metropolis(const InputParameters & parameters)
+  : Sampler(parameters),
+    VectorPostprocessorInterface(this),
+    _distribution_names(getParam<std::vector<DistributionName>>("distributions")),
+    _inputs_names(getParam<std::vector<std::string>>("inputs_names")),
+    _proposal_std(getParam<std::vector<Real>>("proposal_std")),
+    _initial_values(getParam<std::vector<Real>>("initial_values")),
+    _step(getCheckedPointerParam<FEProblemBase *>("_fe_problem_base")->timeStep()),
+    _perf_compute_sample(registerTimedSection("computeSample", 4))
+{
+  for (const DistributionName & name : _distribution_names)
+    _distributions.push_back(&getDistributionByName(name));
+
+  if (_distributions.size() != _inputs_names.size() ||
+      _distributions.size() != _proposal_std.size() ||
+      _distributions.size() != _initial_values.size() ||
+      _inputs_names.size() != _proposal_std.size() ||
+      _inputs_names.size() != _initial_values.size() ||
+      _proposal_std.size() != _initial_values.size())
+    paramError("distributions",
+               "The number of distributions, input names, proposal standard deviations, and "
+               "initial values should be equal");
+
+  setNumberOfRows(getParam<dof_id_type>("num_rows"));
+  setNumberOfCols(_distributions.size());
+  _proposed_sample = 0.0;
+  _acceptance_ratio = 0.0;
+}
+
+void
+Metropolis::sampleSetUp()
+{
+  _values_distributed = isVectorPostprocessorDistributed("inputs_vpp");
+  _inputs_ptr.resize(_inputs_names.size() + 1);
+  for (unsigned int i = 0; i < _inputs_names.size(); ++i)
+  {
+    _inputs_ptr[i] =
+        getVectorPostprocessorValue("inputs_vpp", _inputs_names[i], !_values_distributed);
+  }
```

Comment:
Yes, I think this needs to happen for every compute sample call since the previous sample has changed. So, I need to access the latest previous sample to propose a new sample.

Reply:
Since `_input_ptr` is a pointer now, the variable should be updated automatically when the VPP updates the data.